src/pages/index.js

```diff
@@ -0,0 +1,359 @@
+
+  let cardToDelete = null;
+
+  function openModal(modal) {
+    if (!modal) return;
+    modal.classList.add("modal_opened");
+    document.addEventListener("keydown", handleEscapeKey);
+  }
+
+  function closeModal(modal) {
+    if (!modal) return;
+    modal.classList.remove("modal_opened");
+    document.removeEventListener("keydown", handleEscapeKey);
+  }
+
+  function handleEscapeKey(event) {
+    if (event.key === "Escape") {
+      const openModal = document.querySelector(".modal_opened");
+      if (openModal) closeModal(openModal);
+    }
+  }
+
+  closeModalButtons.forEach((button) => {
+    const modal = button.closest(".modal");
+    button.addEventListener("click", () => closeModal(modal));
+  });
+
+  if (closeDeleteModalButton) {
+    closeDeleteModalButton.addEventListener("click", () => {
+      console.log("Close (X) button clicked in delete modal!");
+      closeModal(deleteCardModal);
+    });
+  }
+
+  if (cancelDeleteButton) {
+    cancelDeleteButton.addEventListener("click", () => {
+      console.log("Cancel button clicked in delete modal!");
+      closeModal(deleteCardModal);
+    });
+  }
+
+  if (deleteCardModal) {
+    deleteCardModal.addEventListener("mousedown", (event) => {
+      if (event.target === deleteCardModal) {
+        closeModal(deleteCardModal);
+      }
+    });
+  }
+
+  api
+    .getUserInfo()
+    .then((userData) => {
+      profileName.textContent = userData.name || "Bessie Coleman";
+      profileDescription.textContent = userData.about || "Civil Aviator";
+      profileAvatar.src = userData.avatar.includes("placeholder")
+        ? "./images/avatar.jpg"
+        : userData.avatar;
+    })
+    .catch((error) => showErrorMessage(error, "Loading profile"));
+
+  function renderCard(data) {
+    const cardElement = createCardElement(data);
+    cardsContainer.prepend(cardElement);
+  }
+
+  function renderInitialCards(cards) {
+    cards.forEach(renderCard);
+  }
+
+  api
+    .getInitialCards()
+    .then((cards) => {
+      console.log("Loaded Cards:", cards);
+      if (!Array.isArray(cards)) {
+        console.error("Error: Expected an array but received:", cards);
+        return;
+      }
+      renderInitialCards(cards);
+    })
+    .catch(console.error);
+
+  function createCardElement(data) {
+    const cardElement = cardTemplate.cloneNode(true).firstElementChild;
+    const cardTitle = cardElement.querySelector(".card__title");
+    const cardImage = cardElement.querySelector(".card__img");
+    const deleteButton = cardElement.querySelector(".card__delete-button");
+    const likeButton = cardElement.querySelector(".card__like-button");
+    const likeCounter = cardElement.querySelector(".card__like-counter"); // Added this line
+
+    cardTitle.textContent = data.name;
+    cardImage.src = data.link;
+    cardImage.alt = data.name;
+
+    const likedCards = JSON.parse(localStorage.getItem("likedCards")) || {};
```

Comment:
Please, delete `localStorage` usage from your project. It can't be used for liking cards because it will work only in one browser. The other computers will not have this info about likes. You should use only `isLiked` from the card data which comes from the server